flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java

```diff
@@ -145,10 +146,29 @@
         }
 
         Configuration configuration = Configuration.fromMap(confMap);
-        if (job.getUpgradeMode() == UpgradeMode.LAST_STATE
-                && !FlinkUtils.isKubernetesHAActivated(configuration)) {
-            return Optional.of(
-                    "Job could not be upgraded with last-state while Kubernetes HA disabled");
+        if (job.getUpgradeMode() == UpgradeMode.LAST_STATE) {
+            if (!FlinkUtils.isKubernetesHAActivated(configuration)) {
+                return Optional.of(
+                        "Job could not be upgraded with last-state while Kubernetes HA disabled");
+            } else if (!configuration.containsKey(
+                            ExecutionCheckpointingOptions.CHECKPOINTING_INTERVAL.key())
+                    || configuration
+                                    .get(ExecutionCheckpointingOptions.CHECKPOINTING_INTERVAL)
+                                    .compareTo(
+                                            configuration.get(
+                                                    ExecutionCheckpointingOptions
+                                                            .MIN_PAUSE_BETWEEN_CHECKPOINTS))
```

Comment:
I dont really understant the point of this check. Is this necessary?

Reply:
@gyfora, this check refers to `CheckpointConfig#setCheckpointInterval` and is necessary. @wangyang0918 WDYT?